.circleci/config.yml

```diff
@@ -266,6 +272,76 @@ jobs:
             docker login --username "${DOCKER_USER}" --password "${DOCKER_PASSWORD}"
             ./gradlew --no-daemon --parallel "-Pbranch=${CIRCLE_BRANCH}" dockerUpload
 
+  extractAPISpec:
+    executor: medium_executor
+    steps:
+      - prepare
+      - attach_workspace:
+          at: ~/project
+      - run:
+          name: Build runnable Teku binaries
+          command: |
+            ./gradlew --no-daemon --parallel installDist
+      - run:
+          name: Extract OpenAPI JSON
+          command: |
+            build/install/teku/bin/teku --network=schlesi --rest-api-docs-enabled=true &
+            TEKU_PID=$!
+            wget -O openapi.json http://localhost:5051/swagger-docs
+            kill $TEKU_PID
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - ./
+
+  publishAPIDoc:
+    executor: node_executor
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/project
+      - add_ssh_keys:
+          fingerprints:
+            - '45:36:58:67:43:15:c6:5f:2c:58:ec:7f:71:e2:e6:ef'
+      - run:
+          name: Set Git user params
+          command: |
+            git config --global user.name $CIRCLE_USERNAME
+            git config --global user.email "${CIRCLE_USERNAME}@users.noreply.github.com"
+      - run:
+          name: Chechout GitHub Pages branch
+          command: |
+            readonly BRANCH='gh-pages'
+            git checkout "$BRANCH" || git checkout -b "$BRANCH"
+            # pull might fail if the branch was just created
+            git pull || true
```

Comment:
@benjamincburns you wrote these lines, do you have any comments on this please?

Reply:
I agree with the suggestion. @ajsutton's suggestion improves on mine in that it'll grab any changes that might've been merged to `gh-pages` between when the build started and when we actually hit this build step. You likely won't see this happen all that often in practice, but I'm sure that you'll see it happen at least a few times, so this will likely prevent you from having to rerun the build.

Reply:
Ignore my last (deleted) comment - I had my wires crossed.

I think @ajsutton's suggestion is functionally equivalent to the code that we already have, but it'll save some noise in the build log if you change line 315 to what he has, and delete lines 316 and 317.

Also if the `git pull` fails in the one-liner that he suggested, it'll cause the `git checkout -b "$BRANCH"` statement to execute, which will also fail, causing the entire build to fail, however the build log might make the reason for the failure a little obscure. A better approach overall would be replace lines 315:317 with something like 

```bash
if [ git branch | grep -q "$BRANCH" ]; then
  git pull
else
  git checkout -b "$BRANCH"
fi
```

But this is likely overkill and I think any of the suggestions will work fine.